File: leetcode/n-repeated-element-in-size-2n-array.py
class Solution(object):
    def repeatedNTimes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        seen = set()

        for num in nums:
            if num in seen:
                return num
            seen.add(num)
        return -1
    
if __name__ == "__main__":
    solver = Solution()

    def run_test(case_num, nums, expected):
        result = solver.repeatedNTimes(nums)
        status = "PASS" if result == expected else "FAIL"
        print(f"test case {case_num}: {result} (expected: {expected}) -> {status}")

        # Example 1
    # n=2, len=4. Target is 3.
    nums1 = [1, 2, 3, 3]
    run_test(1, nums1, 3)
    
    # Example 2
    # n=3, len=6. Target is 2.
    nums2 = [2, 1, 2, 5, 3, 2]
    run_test(2, nums2, 2)
    
    # Example 3
    # n=4, len=8. Target is 5.
    nums3 = [5, 1, 5, 2, 5, 3, 5, 4]
    run_test(3, nums3, 5)

File: golang/repeated-n-times/n-repeated-element-in-size-2n-array.go
package main

import "fmt"

func repeatedNTimes(nums []int) int {
	seen := make(map[int]bool)

	for _, num := range nums {
		if seen[num] {
			return num
		}
		seen[num] = true
	}

	return -1
}

func main() {
	runTest := func(testNum int, nums []int, expected int) {
		result := repeatedNTimes(nums)
		status := "FAIL"
		if result == expected {
			status = "PASS"
		}
		fmt.Printf("Test Case %d: %d (Expected: %d) -> %s\n", testNum, result, expected, status)
	}

	// Example 1
	nums1 := []int{1, 2, 3, 3}
	runTest(1, nums1, 3)

	// Example 2
	nums2 := []int{2, 1, 2, 5, 3, 2}
	runTest(2, nums2, 2)

	// Example 3
	nums3 := []int{5, 1, 5, 2, 5, 3, 5, 4}
	runTest(3, nums3, 5)
}
